.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,52 @@
+name: Publish image to GitHub Container Registry
+
+on:
```

Comment:
I think this workflow should either run the tests itself or be triggered a successful run of the test workflow. WDYT?

Reply:
I agree with this. No sense in building an image from a commit where tests are failing. Question would be: Do we want to chain this to run right after e2e, or do we want it to build it only by running unit testing?

Reply:
My preference would be have a CI job that does all our checks: lint, unit test, E2E test, test Docker build, etc. then triggers this workflow if it all succeeds.

Reply:
I've taken the workflows from `portal-middleware` to enhance this.